File: sol/1524/1524.cpp
struct Solution {
    static constexpr int m = 1e9 + 7;
    int numOfSubarrays(vector<int>& arr) {
        int e = 0, o = 0, res = 0;
        for (int x : arr){
            ++e;
            if (x & 1){
                swap(e, o);
            }
            res = (res + o) % this->m;
        }
        return res;
    }
};

File: sol/1524/1524.py
class Solution:
    m = 10 ** 9 + 7
    def numOfSubarrays(self, arr: List[int]) -> int:
        e, o = 0, 0
        res = 0
        for x in arr:
            e += 1
            if x & 1:
                e, o = o, e
            res += o
        return res % self.m
